Feature/step1

yeonhyunglim opened the pull request:
@SardineTa23
「docker compose up -d」をしたからか、ファイルやフォルダがたくさんできて、ちょっと不便だと感じています。
これらのファイルやフォルダって必要ですか。どうして片づけたらいいのかお聞きしたいです。

yeonhyunglim committed:
DockerでPHP環境構築しました。

yeonhyunglim committed:
docker compose downしてdocker compose up --buildしました。

yeonhyunglim committed:
忘れ物

iwashi623 commented on docker-compose.yml:
```diff
@@ -0,0 +1,37 @@
+    container_name: mysql
+    image: mysql:5.7
+    volumes:
+      - ./db:/var/lib/mysql
```
ここがバインドマウントになっているのが、ファイルが大量生成されている原因かと思います。

iwashi623 reviewed (commented):
```yml
version: '3'

volumes:
  db-store:

services:
  #MySQL用コンテナ
  mysql:
    container_name: mysql
    image: mysql:5.7
    volumes:
      - db-store:/var/lib/mysql
    ports:
      - 3307:3306
    environment:
      MYSQL_ROOT_USER: root
      MYSQL_ROOT_PASSWORD: m1ndfulness!

  #phpMyAdmin用コンテナ
  phpmyadmin:
    container_name: phpMyAdmin
    depends_on:
      - mysql
    image: phpmyadmin/phpmyadmin
    environment:
      PMA_HOST: mysql
    restart: always
    ports:
      - '80:80'

  #php用コンテナ
  php:
    container_name: php-apache
    build: ./php
    volumes:
      - ./html:/var/www/html
    ports:
      - 8080:80
    depends_on:
      - mysql
```

のように書くとバインドマウントからVolume参照に切り替わるので、ローカル内にファイル生成がされずに環境を取り扱えます！

yeonhyunglim committed:
要らないバックグラウンドファイルを削除しました。

yeonhyunglim committed:
volumesを書き換えました。

yeonhyunglim committed:
editorsとarticlesテーブルを作成しました。

iwashi623 commented on docker-compose.yml:
```diff
@@ -0,0 +1,40 @@
+    container_name: mysql
+    image: mysql:5.7
+    volumes:
+      - ./db:/docker-entrypoint-initdb.d
```
申し訳ありません。前回のレビューの時に少しミスしていました😥
` - db-store:/var/lib/mysql`に書き換えるとVolumesで定義したDocker内のVolumeにMySQLのデータが保存されて、ホスト側（自身のPCのローカル環境）から見えない形になります！

iwashi623 reviewed (commented):
レビューで指摘されたとおり書き直したあと、
```
docker-compose down
docker-compose build --no-cache
docker-compose up -d
```
を実行してみてください。それで以前と同じように余計なファイルが生成されなければ修正成功です！

アプリケーション完成まで頑張りましょう👍

yeonhyunglim committed:
RDB設計研修で設計したテーブルのうち、必要なテーブルをローカル環境のMySQLに作成しました。

yeonhyunglim committed:
CREATE DATABASE article_service;を削除しました。
（queryでarticle_serviceいう名前のデータベースが既に作られているため、docker-compose.ymlでMYSQL_DATABASE: article_serviceをするとぶつかってしまい、docker imageを作ることができなくなる。）

yeonhyunglim committed:
投稿フォームのプロントを実装しました。

yeonhyunglim committed:
test

yeonhyunglim committed:
フォルダの名前をhtmlからsrcに変えました。

yeonhyunglim committed:
forbiddenの問題で追加しました。

yeonhyunglim committed:
削除しました。

yeonhyunglim committed:
srcに変更しました。

yeonhyunglim committed:
editor_idがなくて、投稿フォームにデータをmysql articlesテーブルに入れることができなくて、臨時に追加しました。

yeonhyunglim committed:
tagsテーブルにデータをINSERTするqueryを作成しました。

yeonhyunglim committed:
selectを作ったけど、まだmultiple selectができないです。

yeonhyunglim committed:
bootstrapを利用して、dropdown型にselectを変えました。

yeonhyunglim committed:
bootstrap上で、CSSが重ねる問題を解決ため、style.cssを追加しました。

yeonhyunglim committed:
option valueを変えました。

yeonhyunglim committed:
bootstrapを書き換えました。

yeonhyunglim committed:
select option valueを直接手で書かないで、tagsテーブルから習得する必要があったんですね、、、

yeonhyunglim committed:
create.phpを作成して、index.phpからHTMLを移しました。
